sdk/core/core/inc/az_result.h

```diff
@@ -54,7 +54,7 @@ enum
 #define AZ_RETURN_IF_NOT_ENOUGH_SIZE(span, required_space) \
   do \
   { \
-    if (az_span_size(span) < required_space) \
+    if (az_span_size(span) < required_space || required_space < 0) \
```

Comment:
Nit: required_space < 0 is the cheaper check.  I'd do it first

Reply:
Shouldn't required_space (which I'd rename to required_size) being <0 be tested as a precondition - this would reflect a hard/non-recoverable bug.

Reply:
> required_space < 0 is the cheaper check. 

Interesting. That may be true, but it should almost never be true so I am not sure how useful changing the order will be.

> Shouldn't required_space (which I'd rename to required_size) being <0 be tested as a precondition - this would reflect a hard/non-recoverable bug.

I don't think it can be a precondition because it isn't about user defined input parameters. The macro expands to just validate local expression calculations and based on discussions with Cris, this is needed even when preconditions are off.

Reply:
required_space < 0 could means write in reverse 🍔 🥵